git-branchless-record/tests/test_record.rs

```diff
@@ -13,22 +14,31 @@ fn test_record_unstaged_changes() -> eyre::Result<()> {
     git.commit_file("test1", 1)?;
     git.write_file_txt("test1", "contents1\n")?;
     {
-        let (stdout, _stderr) = git.branchless("record", &["-m", "foo"])?;
+        let (stdout, _stderr) = git.branchless("record", &["-m", "foo", "-m", "bar"])?;
         insta::assert_snapshot!(stdout, @r###"
-        [master 914812a] foo
+        [master 872eae1] foo
          1 file changed, 1 insertion(+), 1 deletion(-)
         "###);
     }
 
     {
         let (stdout, _stderr) = git.run(&["show"])?;
-        insta::assert_snapshot!(stdout, @r###"
-        commit 914812ae3220add483f11d851dc59f0b5dbdeaa0
+        // For the empty lines between paragraphs, git show prints 4 spaces
+        // instead of just an empty line. (ie "    \n" instead of "\n") It's
+        // hard to work with this in an inline snapshot, as editors often trim
+        // trailing whitespace, so here we just modify the output (by replacing
+        // any such lines) to make it easier to assert against with insta
+        let whitespace_only_line_regex = Regex::new(r"(?m)^ {4}$").unwrap();
+        let trimmed_stdout = whitespace_only_line_regex.replace_all(&stdout, "");
```

Comment:
This feels wrong, but I couldn't figure out how else to do it. I looked around a bit and didn't see anywhere else that we check snapshots for multiparagraph commit messages from `git show`. In `test_reword`, we look at `git log` output, which doesn't seem to suffer the same behavior.

Reply:
This is fine, and ends up being necessary for snapshot testing in various cases. Can you actually move it into this function?

https://github.com/arxanas/git-branchless/blob/81e298dad0bd0335c4d330816c97fab38a24524a/git-branchless-lib/src/testing.rs#L96-L133

Reply:
Yes, done. I adjusted the comment now that it's more general and less specific to that one test, and adjusted the code slightly to match the style of the other regex in that fn.